instaloader/structures.py

```diff
@@ -150,10 +152,11 @@ def __hash__(self) -> int:
     def _obtain_metadata(self):
         if not self._full_metadata_dict:
             pic_json = self._context.graphql_query(
-                '2b0673e0dc4580674a88d426fe00ea90',
+                'a9441f24ac73000fa17fe6e6da11d59d',
                 {'shortcode': self.shortcode}
             )
             self._full_metadata_dict = pic_json['data']['shortcode_media']
+            self._node['is_loaded_graphsidecar'] = '1'
```

Comment:
What is this needed for?

Reply:
That's necessary because the iterator's `Post` object is not the full `Post` object loaded by the post url. Only in the latter case the Post constructor shall overwrite the `_full_metadata_dict` as would have been done by a download.

Reply:
Would anything bad happen if we did not overwrite the `_full_metadata_dict`?

Reply:
What do you mean? When the post is deserialized from disk, then I recognize it in the constructor and overwrites the `_full_metadata_dict` just as if a download just had happened. Of course, for a download `_full_metadata_dict` is overwritten in `_obtain_metadata` as before. Without doing that, `_field`/`_full_metadata` could trigger a new (unwanted) download.

Reply:
Ah I see. Thanks for sharing your observation!

f85e49d521ca6a838c23213b42acc8c83e9a7e6e (currently only in master) fixes that unnecessary fetch triggered in `get_sidecar_nodes()`. This should also make your change in `Post._obtain_metadata()` and `Post.__init__()` unnecessary.

`_full_metadata` triggers the metadata fetch on a Post created with `load_structure_from_file()`; however, `_full_metadata` is not supposed to be called unless we can be sure that the required data is not available in `_node`. `_field()` does not do a full metadata fetch if the required data is available in `_node`.